tests/configurable_cluster.py

```diff
@@ -140,6 +149,11 @@ def get_down_hostname(self, host_name):
 
     def exec_cmd_on_host(self, host, cmd, user=None, raise_error=True,
                          tty=False):
+        # Ensure command is invoked with sudo since EMR's login user is ec2-user
+        # If sudo is already present in the command then no error will occur
+        # as arbitrary nesting of sudo is allowed
```

Comment:
touch "foo "'"'" bar"
bash -c "foo "'"'" bar"

I think you need to test both.